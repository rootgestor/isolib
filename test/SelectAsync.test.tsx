import React from 'react';
import { SelectAsync } from '../dist/index';
// const { render } = require('@testing-library/react');
const ShallowRenderer = require('react-test-renderer/shallow');

const args = {
  mode: 'multiple',
  placeholder: 'Select users',
  style: { width: '100%' },
  defaultValue: ['asdj'],
  defaultOptions: [{ label: 'sasasdss', value: 'asdj' }],
};

describe('SelectAsync', () => {
  const renderer = new ShallowRenderer();

  it('render snapshot', () => {
    const selectAsync = renderer.render(<SelectAsync {...args} />);
    expect(selectAsync).toMatchSnapshot();
  });

  // it('renders without crashing', () => {
  //   const { container } = render(<Breadcrumb {...args} />);
  //   expect(container).toBe(true);
  // });
});
