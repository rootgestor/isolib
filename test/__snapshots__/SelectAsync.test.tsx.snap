// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectAsync render snapshot 1`] = `
<ForwardRef(InternalSelect)
  filterOption={false}
  labelInValue={true}
  loading={false}
  mode="multiple"
  onChange={[Function]}
  onFocus={[Function]}
  onSearch={[Function]}
  options={
    Array [
      Object {
        "label": "sasasdss",
        "value": "asdj",
      },
    ]
  }
  placeholder="Select users"
  style={
    Object {
      "width": "100%",
    }
  }
  value={
    Array [
      "asdj",
    ]
  }
/>
`;
